kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriberServiceImpl.java

```diff
@@ -362,4 +366,21 @@ public void deleteAllowed(Subscriber subscriber) {
             subscriptionService.deletePreconditionCheck(subscription);
         }
     }
+
+    @Override
+    public void deactivateAllSubscriptionsForSubscriber(long callingNumber) {
+        LOGGER.info("Recieved Release Number {} for Deactivation.", callingNumber);
+        Subscriber subscriberByMsisdn = this.getSubscriber(callingNumber);
+        if (subscriberByMsisdn == null) {
+            LOGGER.info("Subscriber for msisdn {} is not found." , callingNumber);
+            return;
+        }
+        LOGGER.info("Found Subscriber for msisdn {} .", callingNumber);
+        for (Subscription subscription : subscriberByMsisdn.getAllSubscriptions()) {
+            LOGGER.info("Deactivating Subscrition with start date {} for msisdn .", subscription.getStartDate());
+            subscriptionService.deactivateSubscription(subscription, DeactivationReason.MOHFW_REQUEST);
+            LOGGER.info("Deactivated Subscrition with start date {} for msisdn .", subscription.getStartDate());
```

Comment:
Add deactivating subscription add subscription phone number and start date and current timeto audit service .